src/pages/index.js

```diff
@@ -1,93 +1,236 @@
 
 //define global variables
 const profileModal = "#profile-modal";
+const profileImgModal = "#edit-profile-image-modal";
 const profileName = ".profile__name";
 const profileDescription = ".profile__description";
+const profileImage = ".profile__image";
+const profileImageEl = document.querySelector(profileImage);
 const profileEditButton = document.querySelector(".profile__edit-button");
+const profileEditImageButton = document.querySelector(
+  ".profile__image__edit-button"
+);
 const addCardButton = document.querySelector(".profile__add-button");
 const addModal = "#add-modal";
+const editProfileImgForm = document.forms["edit-profile-image-form"];
 const addCardForm = document.forms["add-card-form"];
 const previewModal = "#preview-modal";
+const deleteModal = "#delete-modal";
 
-//add card section to DOM & add initial cards
+const api = new Api({
+  baseUrl: apiConfig.baseUrl,
+  headers: apiConfig.headers,
+});
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", imageClick);
-  return card.getView();
-}
+// Create user info instance
+const userInfo = new UserInfo({
+  userName: profileName,
+  userJob: profileDescription,
+});
+
+//receive user info
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      userName: res.name,
+      userJob: res.about,
+    });
+
+    profileImageEl.setAttribute("src", res.avatar);
+  })
+  .catch((err) => {
+    console.error("Failed to receive user info", err);
+  });
+
+//add card section to DOM & add initial cards
 
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (cardData) => {
+    renderer: (cardData, method) => {
+      if (cardData.link) {
+        cardData.cardLink = cardData.link;
+      }
+      if (cardData.name) {
+        cardData.cardName = cardData.name;
+      }
+
       const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement, "append");
+      cardSection.addItem(cardElement, method);
     },
   },
   ".cards__list"
 );
 
-cardSection.renderItems();
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards, "append");
+  })
+  .catch((err) => {
+    console.error("Failed to receive card data", err);
+  });
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    imageClick,
+    deleteClick,
+    likeClick
+  );
+  return card.getView();
+}
 
 //Submit handling functions
 
-function handleProfileFormSubmit(input) {
-  userInfo.setUserInfo(input);
-  profileEditPopup.close();
+function handleProfileFormSubmit(input, button) {
+  api
+    .setUserInfo(input)
+    .then((data) =>
+      userInfo.setUserInfo({ userName: data.name, userJob: data.about })
+    )
+    .catch((err) => {
+      console.error("Failed to save changes in user info", err);
+    })
+    .finally(() => {
+      profileEditPopup.close();
+      button.textContent = "Save";
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
This is very interesting, but it's a bit too much for me right now 🙏